src/Type/Constant/ConstantArrayType.php

```diff
@@ -386,6 +396,17 @@ public function findTypeAndMethodName(): ?ConstantArrayTypeAndMethod
 			if ($this->isOptionalKey(0) || $this->isOptionalKey(1)) {
 				$has = $has->and(TrinaryLogic::createMaybe());
 			}
+			if ($scope === null) {
+				$scope = new OutOfClassScope();
+			}
```

Comment:
I wasn't sure if this should be pushed to the top of the method – to provide a value for a default nullable meant for backward compatibility. Or instantiated when it's finally needed.